backend-rust/src/block_special_event.rs

```diff
@@ -0,0 +1,421 @@
+    #[graphql(
+        name = "_",
+        deprecation = "Don't use! This field is only in the schema to make this a valid GraphQL \
+                       type (which does not allow types without any fields)"
+    )]
+    dummy: bool,
+}
+
+#[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+struct BakerPoolRewardTarget {
+    baker_id: BakerId,
+}
+
+#[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+struct AccountAddressAmount {
+    account_address: AccountAddress,
+    amount:          Amount,
+}
+
+impl From<(sdk_id_types::AccountAddress, sdk_common_types::Amount)> for AccountAddressAmount {
+    fn from((address, amount): (sdk_id_types::AccountAddress, sdk_common_types::Amount)) -> Self {
+        Self {
+            account_address: address.to_owned().into(),
+            amount:          amount.micro_ccd().into(),
+        }
+    }
+}
+
+#[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+pub struct ValidatorSuspended {
+    baker_id: BakerId,
+    account:  AccountAddress,
+}
+
+#[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+pub struct ValidatorPrimedForSuspension {
+    baker_id: BakerId,
+    account:  AccountAddress,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, sqlx::Type, async_graphql::Enum)]
+#[sqlx(type_name = "special_transaction_outcome_type")]
+pub enum SpecialEventTypeFilter {
+    Mint,
+    FinalizationRewards,
+    BlockRewards,
+    BakingRewards,
+    PaydayAccountReward,
+    BlockAccrueReward,
+    PaydayFoundationReward,
+    PaydayPoolReward,
+    ValidatorSuspended,
+    ValidatorPrimedForSuspension,
+}
+
+impl From<&concordium_rust_sdk::types::SpecialTransactionOutcome> for SpecialEventTypeFilter {
+    fn from(value: &concordium_rust_sdk::types::SpecialTransactionOutcome) -> Self {
+        use concordium_rust_sdk::types::SpecialTransactionOutcome as O;
+        match value {
+            O::BakingRewards {
+                ..
+            } => Self::BakingRewards,
+            O::Mint {
+                ..
+            } => Self::Mint,
+            O::FinalizationRewards {
+                ..
+            } => Self::FinalizationRewards,
+            O::BlockReward {
+                ..
+            } => Self::BlockRewards,
+            O::PaydayFoundationReward {
+                ..
+            } => Self::PaydayFoundationReward,
+            O::PaydayAccountReward {
+                ..
+            } => Self::PaydayAccountReward,
+            O::BlockAccrueReward {
+                ..
+            } => Self::BlockAccrueReward,
+            O::PaydayPoolReward {
+                ..
+            } => Self::PaydayPoolReward,
+            O::ValidatorSuspended {
+                ..
+            } => Self::ValidatorSuspended,
+            O::ValidatorPrimedForSuspension {
+                ..
+            } => Self::ValidatorPrimedForSuspension,
+        }
+    }
+}
+
+fn connection_from_slice<A: async_graphql::OutputType>(
```

Comment:
This does not have to be specific to blocks.

I should use this from the nodes collector querying.

Reply:
I like this function. Gj